spec/app/domain/authentication/resource_restrictions/get_restriction_from_annotation_spec.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe(Authentication::ResourceRestrictions::GetRestrictionFromAnnotation) do
+  let(:authenticator_name) { 'authn-dummy' }
+  let(:service_name) { 'dummy_service' }
+  let(:restriction_name) { 'user_email' }
+  let(:general_annotation_name) { "#{authenticator_name}/#{restriction_name}" }
+  let(:service_specific_annotation_name) { "#{authenticator_name}/#{service_name}/#{restriction_name}" }
+
+  #  ____  _   _  ____    ____  ____  ___  ____  ___
+  # (_  _)( )_( )( ___)  (_  _)( ___)/ __)(_  _)/ __)
+  #   )(   ) _ (  )__)     )(   )__) \__ \  )(  \__ \
+  #  (__) (_) (_)(____)   (__) (____)(___/ (__) (___/
+
+  context "GetRestrictionFromAnnotation" do
+    context "Annotation is general" do
+      subject do
+        Authentication::ResourceRestrictions::GetRestrictionFromAnnotation.new.call(
+          annotation_name: general_annotation_name,
+          authenticator_name: authenticator_name,
+          service_id: nil
+        )
+      end
+
+      it "returns restriction name and true" do
+        restriction_name, is_general_restriction = subject
+        expect(restriction_name).to eql(restriction_name)
```

Comment:
look strange how do you know what you compared the names of the input and the output are similar

Reply:
Fixed